presto-orc/src/test/java/com/facebook/presto/orc/TestStreamLayout.java

```diff
@@ -108,4 +135,201 @@ public void testByColumnSize()
 
         assertFalse(iterator.hasNext());
     }
+
+    @Test
+    public void testByStreamSizeStreamOrdering()
+    {
+        List<StreamDataOutput> streams = createStreams();
+        ByStreamSize streamLayout = new ByStreamSize();
+        StreamOrderingLayout streamOrderingLayout = new StreamOrderingLayout(createStreamReorderingInput(), streamLayout);
+        streamOrderingLayout.reorder(streams, createNodeIdToColumnId(), createColumnEncodings());
+
+        Iterator<StreamDataOutput> iterator = streams.iterator();
+        verifyFlatMapColumns(iterator);
+        // non flat map columns
+        verifyStream(iterator.next().getStream(), 5, 0, StreamKind.DATA, 1);
+        verifyStream(iterator.next().getStream(), 2, 0, StreamKind.LENGTH, 2);
+        verifyStream(iterator.next().getStream(), 1, 0, StreamKind.DATA, 3);
+        verifyStream(iterator.next().getStream(), 4, 0, StreamKind.LENGTH, 5);
+        verifyStream(iterator.next().getStream(), 3, 0, StreamKind.DATA, 8);
+        verifyStream(iterator.next().getStream(), 1, 0, StreamKind.PRESENT, 12);
+        assertFalse(iterator.hasNext());
+    }
+
+    @Test
+    public void testByColumnSizeStreamOrdering()
+    {
+        List<StreamDataOutput> streams = createStreams();
+        ByColumnSize streamLayout = new ByColumnSize();
+        StreamOrderingLayout streamOrderingLayout = new StreamOrderingLayout(createStreamReorderingInput(), streamLayout);
+        streamOrderingLayout.reorder(streams, createNodeIdToColumnId(), createColumnEncodings());
+
+        Iterator<StreamDataOutput> iterator = streams.iterator();
+        verifyFlatMapColumns(iterator);
+        // non flat map columns
+        verifyStream(iterator.next().getStream(), 5, 0, StreamKind.DATA, 1);
+        verifyStream(iterator.next().getStream(), 2, 0, StreamKind.LENGTH, 2);
+        verifyStream(iterator.next().getStream(), 4, 0, StreamKind.LENGTH, 5);
+        verifyStream(iterator.next().getStream(), 3, 0, StreamKind.DATA, 8);
+        verifyStream(iterator.next().getStream(), 1, 0, StreamKind.DATA, 3);
+        verifyStream(iterator.next().getStream(), 1, 0, StreamKind.PRESENT, 12);
+
+        assertFalse(iterator.hasNext());
+    }
+
+    private static Map<Integer, ColumnEncoding> createColumnEncodings()
+    {
+        SortedMap<Integer, DwrfSequenceEncoding> seqIdToEncodings1 = ImmutableSortedMap.of(
```

Comment:
Looks like right now there is a full overlap of keys in the streams and in the ordering config. 
```
100,200,300 -> 100,200,300
100,200,300,400 -> 100,200,300,400,500
```

Can you please modify this to have a key in the streams that is not present in the ordering config?

Reply:
Sure, will add